src/table/table.scss

```diff
@@ -122,265 +251,316 @@
-          justify-content: flex-end !important;
+    // #region Selection
+    &:not(.iui-selected) + .iui-selected {
+      padding-bottom: $iui-xxs;
+      border-bottom: none;
+    }
 
-          > .iui-icon-wrapper {
-            cursor: pointer;
+    &.iui-selected {
+      padding: 0;
+      border-bottom: none;
+      @include themed {
+        border: solid 1px t(iui-color-foreground-primary);
+        background: rgba(t(iui-color-foreground-primary-rgb), t(iui-opacity-5));
+      }
 
-            > .iui-icon {
-              width: $iui-icons-default;
-              height: $iui-icons-default;
-            }
-          }
+      &:hover {
+        @include themed {
+          background: rgba(
+            t(iui-color-foreground-primary-rgb),
+            t(iui-opacity-5)
+          );
         }
+      }
 
-        > .iui-filter,
-        > .iui-sort,
-        > .iui-sort > .iui-icon-wrapper {
-          height: $iui-icons-default;
-          width: $iui-icons-default;
-          display: flex;
-          justify-content: center;
-          align-items: center;
-          visibility: hidden;
+      + .iui-selected {
+        border-bottom: none;
+        padding-bottom: $iui-xxs;
+        padding-top: 0;
+        @include themed {
+          border-top: 1px solid
+            rgba(t(iui-color-foreground-primary-rgb), t(iui-opacity-4));
         }
+      }
 
-        > .iui-filter.iui-active {
-          visibility: visible;
-  
-          > .iui-icon {
-            @include themed {
-              fill: t(iui-icons-color-primary);
-            }
-          }
+      &:last-child {
+        padding-bottom: 1px;
+        @include themed {
+          border-bottom: 1px solid t(iui-color-foreground-primary);
         }
+      }
 
-        &:hover {
-          @include themed {
-            background-color: t(iui-color-background-4);
-          }
-
-          > .iui-filter,
-          > .iui-sort,
-          > .iui-sort > .iui-icon-wrapper {
-            visibility: visible;
-          }
-        }
+      &:first-child {
+        border-bottom: none;
+        padding-bottom: $iui-xxs;
       }
 
-      &.iui-active-sort {
+      // Last selected item in a row.
+      + :not(.iui-selected) {
+        padding-top: 0;
+        margin-left: 0;
+        margin-right: 0;
         @include themed {
-          background-color: t(iui-color-background-4);
+          border-top: 1px solid t(iui-color-foreground-primary);
         }
+      }
 
-        > .iui-sort,
-        > .iui-sort > .iui-icon-wrapper {
-          visibility: visible;
+      // Added by Jon, not fully tested
+      + .iui-expanded-content {
+        @include themed {
+          border-color: t(iui-color-foreground-primary);
         }
       }
     }
 
-    &.iui-tables-row-active {
-      @include themed {
-        border: solid 1px t(iui-color-foreground-primary);
-        background: rgba(t(iui-color-foreground-primary-rgb), t(iui-opacity-5));
-      }
+    &:last-child {
```

Comment:
This is not really for selected row...

Reply:
Yes it is. Unselected row use bottom-border: 1px, padding-bottom: 1px, because iui-body use bottom-border:1px, last-child does not need a border. 
- Unselected use bottom border.
- Selected use top-border.
- Body has a bottom-border.
  - last row does not need bottom-border because of body bottom-border.